### DISCUSSION TITLE
ListBlocks Don't Support Nesting As a Child Node

### DISCUSSION THREAD

**Discussion by secondmanveran** [thephpleague/commonmark#1006] [Q&A]
I'm building out a Container extension for making attractive state based containers, and CodeGroups with tabs for different language variations. 

All runs well until I recently discovered that nested lists get flattened inside another node.

So I have a list inside a container:

````md
::: info requirements

- Endpoint: `/api/2.1/accessories/{id}`
- Method: `GET`
- Params:
  - Allowed: `record id in endpoint url`
  - Required: `record id in endpoint url`

:::
````

but when rendered I get this:

<img width="639" alt="Screenshot 2024-01-03 at 3 18 45 AM" src="https://github.com/thephpleague/commonmark/assets/97000801/01c1d08f-9d5f-46ac-b7f7-82249e1fce2c">

If I move the list out of the container, it renders as expected:

<img width="451" alt="Screenshot 2024-01-03 at 3 21 44 AM" src="https://github.com/thephpleague/commonmark/assets/97000801/fad53a17-91a2-4ca3-9b63-20615d67f929">

I considered using the event system via DocumentParsedEvent to extract the list nodes to re-render them, but the problem is that the block is already parsed as flattened. I thought of using the DocumentPreParsedEvent, but without the nodes to iterate I haven't found a way to get the raw markdown for just the list inside the container. 

Any pointers would be great.

**Comment by colinodell** [comment]
Double-check whether the resulting HTML looks correct (in case this is a CSS issue).

If you confirm that the list is not nested, that works suggest the issue is with how it's being parsed. Double-check whether your custom container parser is accidentally consuming the leading whitespace that the list parser uses to determine if a line belongs to the same list block or a new nested one.
